### DISCUSSION TITLE
Showing page for install location in standard WixStandardBootstrapperApplication

### DISCUSSION THREAD

**Discussion by radz1** [wixtoolset/issues#6320] [Questions] [answered]
Hello, I am using bundle and it contains 2 msi. I need to have option for setting the install location and send pass the InstallLocation variable to both chained msi. I am able to achieve it using the Option dialog and variable as below.

I have 2 query here: 
1. **Can we show this option dialog as next page like wizard**. for example, on click of install button-> set location page will be visible. Is this possible ? if yes pls suggest how can we do.

2. **In the Option[file location] dialog, how can we enable validation for the install directory.** For example, Even if we add invalid path or drive which does not exist, it is not getting validated before clicking install button. 

```
<Bundle Name="2.5.0.32" Version="2.5.0.32" Manufacturer="Inc." UpgradeCode="390a8050-9462-47d5-8617-0e983adc1d88" IconSourceFile="16x16.ico"> 
    <Log PathVariable="c:\temp\packagelog.txt"/>
    <BootstrapperApplicationRef Id="WixStandardBootstrapperApplication.RtfLicense">
      <bal:WixStandardBootstrapperApplication
        xmlns:bal="http://schemas.microsoft.com/wix/BalExtension"
        LicenseFile="Assets\license.rtf"  LogoFile="Assets\logo.png" ThemeFile="Resources\RtfTheme.xml" LocalizationFile="Resources\RtfTheme.wxl" SuppressOptionsUI="no"/>  
        
    </BootstrapperApplicationRef>
```

```
<Variable Name="InstallFolder" Type="string" Value="[ProgramFiles64Folder]" bal:Overridable="yes" />
    
    <Chain>      

      <MsiPackage Id="msi1" SourceFile="MSI\msi1.msi" DisplayInternalUI ="yes" Visible="no">
        <MsiProperty Name="INSTALLFOLDER" Value="[InstallFolder]" />
      </MsiPackage>

      <MsiPackage Id="msi2" SourceFile="msi1.msi" DisplayInternalUI ="yes" Visible="no">
        <MsiProperty Name="PASSWORD" Value="[PASSWORD]" />
        <MsiProperty Name="INSTALLFOLDER" Value="[InstallFolder]" />
      </MsiPackage>

    </Chain>
```

**Comment by rseanhall** [comment, marked as answer]
1. I think the closest you're going to get in v3.x is http://windows-installer-xml-wix-toolset.687559.n2.nabble.com/Re-Always-show-the-option-dialog-P-td7598496.html#a7598499.
2. Not possible today without creating your own Bootstrapper Application - https://github.com/wixtoolset/issues/issues/4676.

**Reply by radz1** [reply]
thanks for the your quick response

**Comment by robmen** [comment]
1. `wixstdba` does not support that today. You could write your own BootstrapperApplication and have complete control of the UI.
2. Again, `wixstdba` does not support this today. A BAFunction could extend the `wixstdba` and validate the value when pressing the install button... but not on the Options page. Again, a custom BootstrapperApplication would be able to have complete control of the UI.

There are some improvements in the UI framework that `wixstdba` uses in WiX v4 that should allow more customization. But WX v4 is still under development so a custom BootstrapperApplication is your best bet for customizing the installation UI.

**Reply by radz1** [reply]
thanks for quick reply
